packages/velo-external-db-core/lib/router.js

```diff
@@ -243,60 +254,83 @@ const createRouter = () => {
+            const schemaIds = await executeSchemaHooksFor(SchemaActions.BeforeFind, schemaPayloadFor(SchemaOperations.FIND, req.body), requestContextFor(SchemaOperations.FIND, req.body)) 
+
             if (schemaIds && schemaIds.length > 10) {
                 throw new InvalidRequest()
             }
             const data = await schemaService.find(schemaIds)
-            res.json(data)
+            const dataAfterAction = await executeSchemaHooksFor(SchemaActions.AfterFind, data, requestContextFor(SchemaOperations.FIND, req.body))
+            res.json(dataAfterAction)
         } catch (e) {
             next(e)
         }
     })
 
     router.post('/schemas/create', async(req, res, next) => {
         try {
-            const { collectionName } = req.body
+            const collectionName  = await executeSchemaHooksFor(SchemaActions.BeforeCreate, schemaPayloadFor(SchemaOperations.CREATE, req.body), requestContextFor(SchemaOperations.CREATE, req.body))
             const data = await schemaService.create(collectionName)
-            res.json(data)
+            
+            const dataAfterAction = await executeSchemaHooksFor(SchemaActions.AfterCreate, data, requestContextFor(SchemaOperations.CREATE, req.body))
+
+            res.json(dataAfterAction)
         } catch (e) {
+            console.log(e)
             next(e)
         }
     })
 
     router.post('/schemas/column/add', async(req, res, next) => {
         try {
-            const { collectionName, column } = req.body
+            const { collectionName } = req.body
+            const column =  await executeSchemaHooksFor(SchemaActions.BeforeColumnAdd, schemaPayloadFor(SchemaOperations.COLUMN_ADD, req.body), requestContextFor(SchemaOperations.COLUMN_ADD, req.body))
+            
             const data = await schemaService.addColumn(collectionName, column)
-            res.json(data)
+
+            const dataAfterAction = await executeSchemaHooksFor(SchemaActions.AfterColumnAdd, data, requestContextFor(SchemaOperations.COLUMN_ADD, req.body))
+
+            res.json(dataAfterAction)
         } catch (e) {
+            console.log(e)
             next(e)
         }
     })
 
     router.post('/schemas/column/remove', async(req, res, next) => {
         try {
-            const { collectionName, columnName } = req.body
+            const { collectionName } = req.body
+            const columnName =  await executeSchemaHooksFor(SchemaActions.BeforeColumnRemove, schemaPayloadFor(SchemaOperations.COLUMN_REMOVE, req.body), requestContextFor(SchemaOperations.COLUMN_REMOVE, req.body))
+            
             const data = await schemaService.removeColumn(collectionName, columnName)
-            res.json(data)
+
+            const dataAfterAction = await executeSchemaHooksFor(SchemaActions.AfterColumnRemove, data, requestContextFor(SchemaOperations.COLUMN_REMOVE, req.body))
+            res.json(dataAfterAction)
         } catch (e) {
             next(e)
         }
```

Comment:
In general, you have a lot of unnecessary code duplication in this file. 
I understand there should be a route per operation, but the handlers can be united. I think you should think about a `routerHandler` function(name TBD of course), which gets the operation handler, before hook and after hook, and will perform the shared logic.
Alternatively you can create a class as well if you think it's more fitting to your solution

Reply:
I'm not sure that I understand what your alternative is.
Could you describe how the `routerHandler` would look and how that would prevent the duplication?